### PR TITLE
Deferred join

### PR DESCRIPTION
**Pull request by samjxn**
#### Summary of Changes

Joining a game now occurs when a client enters a username and presses "Enter Game"

What's left to do:
 - Backend:  Name player some random boxer name if they don't enter a username
 - Pick one:
   - Backend:  If a user does request to join a room after already being assigned one, remove them from their current game and re-assign them
   - Frontend:  Don't let user's join a game if they are already in one

#### Screenshots or Demo (optional)
preview

#### Build Status
TODO:  Replace YOUR_BRANCH_HERE - [Build Status](https://travis-ci.org/jonnykry/boxD)

#### Relevant GIF (How does this PR make you feel?)
preview

### CODE REVIEW

**Comment by samjxn** [comment]
+1 to Jonny's commits

File: server/boxd_api.py [hunk 1]
```diff
@@ -121,7 +117,20 @@ def on_message(self, message_json):
...
+                assigned_game = GameRunner.assign_player(self.client_id)
+                GameRunner.update_player_name(self.client_id, name)
+
+                ConnectionManager.send_message(self.client_id, "You joined game {} as {}".format(assigned_game, name))
+                ConnectionManager.send_message(self.client_id, "The game has the following players:  {}"
+                                               .format(GameRunner().get_other_player_ids(self.client_id)))  # TODO:  Send player scores and board state
+                ConnectionManager.send_to_all(GameRunner.get_other_player_ids(self.client_id),
+                                              "{} (client {}) joined the game".format(name, self.client_id))
+
+            # TODO:  elif message['type'] == 'LEAVE_GAME':
```
Comment (jonnykry) [inline server/boxd_api.py:131]: want to uncomment this and put a `print` placeholder so I can implement it anyways?

(as well as any other placeholders for events you're thinking of)

Comment (samjxn) [reply server/boxd_api.py:131]: I wouldn't call that critical for this PR.  For this PR, it's assumed that a client is in the game until their connection closes

Comment (samjxn) [reply server/boxd_api.py:131]: But yes.

**Pull request merged**